shared-svcs-stamp.bicep

```
@@ -312,103 +316,115 @@ resource frontDoorWafPolicy 'Microsoft.Network/FrontDoorWebApplicationFirewallPo
   }
 }
 
-
-@description('Front Door (classic) to be our global router.')
-resource frontDoor 'Microsoft.Network/frontDoors@2021-06-01' = {
+@description('Front Door Profile (Premium) to be our global router. We use the premium SKU for its support of Private Link and managed WAF rules.')
+resource frontDoorProfile 'Microsoft.Cdn/profiles@2024-02-01' = {
   name: frontDoorName
   location: 'global'
-  properties: {
-    backendPools: [
-      {
-        name: 'MultiClusterBackendPool'
-        properties: {
-          backends: [
-            {
-              address: 'bicycle.cloudapp.azure.com'
-              httpPort: 80
-              httpsPort: 443
-              priority: 1
-              weight: 50
-              enabledState: 'Enabled'
-              backendHostHeader: 'bicycle.cloudapp.azure.com'
-            }
-          ]
-          healthProbeSettings: {
-            id: '${resourceId('Microsoft.Network/frontDoors', frontDoorName)}/HealthProbeSettings/healthProbeSettings'
-          }
-          loadBalancingSettings: {
-            id: '${resourceId('Microsoft.Network/frontDoors', frontDoorName)}/LoadBalancingSettings/loadBalancingSettings'
-          }
+  sku: {
+    name: 'Premium_AzureFrontDoor'
+  }
+
+  resource secPolicies 'securityPolicies' ={
+    name: 'afd-sec-policies'
+    properties: {
+      parameters: {
+        type: 'WebApplicationFirewall'
+        wafPolicy: {
+          id: frontDoorWafPolicy.id
         }
+        associations: [
+          {
+            domains: [
+              {
+                id: endpoint.id
+              }
+            ]
+            patternsToMatch: [
+              '/*'
+            ]
+          }
+        ]
       }
-    ]
-    healthProbeSettings: [
-      {
-        name: 'healthProbeSettings'
-        properties: {
-          intervalInSeconds: 30
-          path: '/favicon.ico'
-          protocol: 'Https'
-          enabledState: 'Enabled'
-          healthProbeMethod: 'HEAD'
+    }
+  }
+
+  resource endpoint 'afdEndpoints' = {
+    name: frontDoorName
+    location: 'global'
+    properties: {
+      autoGeneratedDomainNameLabelScope: 'TenantReuse'
+      enabledState: 'Enabled'
+    }
+
+    resource frontDoorRoute 'routes' = {
+      name: '${frontDoorName}-route'
+      dependsOn: [
+        frontDoorOriginGroup::frontDoorOrigin
```

I suggest adding a comment explaining why we need this explicit dependency:

```suggestion
        frontDoorOriginGroup::frontDoorOrigin // This explicit dependency is required to ensure that the origin group is not empty when the route is created.
```